db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java

```diff
@@ -477,7 +481,9 @@ protected SchemaDescriptor initDefaultSchemaDescriptor() throws StandardExceptio
         */
         if(cachedInitialDefaultSchemaDescr==null){
             DataDictionary dd=getDataDictionary();
-            String authorizationId=getSessionUserId();
+            if (defaultSchema != null) {
+                return dd.getSchemaDescriptor(defaultSchema, getTransactionCompile(), true);
```

Comment:
I think we need to set cachedInitialDefaultSchemaDescr to the returned schema descriptor before returning from the function.